src/transformers/trainer.py

```diff
@@ -1255,6 +1259,10 @@ def train(
 
                     self._maybe_log_save_evaluate(tr_loss, model, trial, epoch)
 
+                ort_step_metrics = speed_metrics("train_step", 
+                    start_train_step_time, self.args.per_device_train_batch_size)
```

Comment:
why does this metric not use `start_train_stable_time`

Reply:
start_train_stable_time is captured just once after the first training loop. This is then used at the end of training to compute average samples/sec excluding the initial setup cost.

train_step captures throughput each training step.

Reply:
also if this is measuring the step time for each step, how is this different from the current metrics, and do we need a specific one for ort, if it is doing so for the first step too?

Reply:
The current metric prints steps/sec, we could do the conversion ourselves by multiplying that with batch size per step. This is just easier for visual debug without having to do separate math.